### DISCUSSION TITLE
Suppressing VTK warnings and logs

### DISCUSSION THREAD

**Discussion by banesullivan** [pyvista/pyvista#3495] [Q&A] [answered]
When running the [`collision` filter](https://docs.pyvista.org/api/core/_autosummary/pyvista.PolyDataFilters.collision.html), sometimes you may see a lot of repeated warnings from VTK like:

```
Generic Warning: In /opt/glr/vtk/vtk-ci-ext/0/Common/Core/vtkMath.cxx, line 814
vtkMath::Jacobi: Error extracting eigenfunctions

Generic Warning: In /opt/glr/vtk/vtk-ci-ext/0/Common/Core/vtkMath.cxx, line 814
vtkMath::Jacobi: Error extracting eigenfunctions
```

There is a way to suppress these warnings which I will share as the answer below

**Comment by banesullivan** [comment]
We can suppress these warnings by piping the VTK log to a file and also turning off the global display:

```py
import pyvista as pv
import vtk

pv.set_error_output_file("errors.txt")  # Pipe VTK warnings to a file
vtk.vtkObject.GlobalWarningDisplayOff()  # Turn off the default error display
```

**Reply by banesullivan** [reply]
cc @LukeDiederich

**Comment by banesullivan** [comment]
Following up about `vtkLogger` output from https://github.com/pyvista/pyvista/issues/3612 as well. To disable these, do:

```py
import vtk
vtk.vtkLogger.SetStderrVerbosity(vtk.vtkLogger.VERBOSITY_OFF)
```

For example on VTK 9.2.2
```py
import vtk
print(vtk.vtkVersion().GetVTKVersion())

vtk.vtkLogger.SetStderrVerbosity(vtk.vtkLogger.VERBOSITY_OFF)

source = vtk.vtkSphereSource()

alg = vtk.vtkExtractEdges()
alg.SetInputConnection(source.GetOutputPort())
alg.Update()
```

**Comment by user27182** [comment, marked as answer]
For `pyvista>=0.45` there is now [pv.vtk_verbosity](https://docs.pyvista.org/api/utilities/_autosummary/pyvista.vtk_verbosity.html):

``` python
import pyvista as pv
with pv.vtk_verbosity('off'):
    # All vtk errors and warnings are silenced in this context
   ...
```
